src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AdjudicationsResourceTest.java

```diff
@@ -605,4 +618,190 @@ public void deleteHearing() {
+                .expectStatus().isCreated()
+                .expectBody()
+                .jsonPath("$.findingCode").isEqualTo(FindingCode.NOT_PROCEED.name())
+                .jsonPath("$.pleaFindingCode").isEqualTo(PleaFindingCode.GUILTY.name());
+        }
+
+        private ResponseSpec createHearingResult(List<String> headers, Map payload, Long adjudicationNumber, Long hearingId) {
+            return webTestClient.post()
+                .uri("/api/adjudications/adjudication/"+adjudicationNumber+"/hearing/"+hearingId+"/result")
+                .headers(setAuthorisation(headers))
+                .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+                .bodyValue(payload)
+                .exchange();
+        }
+    }
+
+    @Nested
+    public class AmendHearingResult {
+        final List<String> valid = List.of("ROLE_MAINTAIN_ADJUDICATIONS");
+        final List<String> invalid = List.of("ROLE_SYSTEM_USER");
+
+        final Map validRequest = Map.of("pleaFindingCode", PleaFindingCode.GUILTY, "findingCode", FindingCode.NOT_PROCEED, "adjudicator", "ITAG_USER");
+        final Map invalidRequest = Map.of("pleaFindingCode", PleaFindingCode.GUILTY, "findingCode", FindingCode.NOT_PROCEED, "adjudicator", "TWRIGHT");
+
+        @Test
+        public void amendHearingResultReturns403ForInvalidRoles () {
+            amendHearingResult(invalid, validRequest, -9L, -1L)
+                .expectStatus().isForbidden();
+        }
+
+        @Test
+        public void amendHearingResultReturns404DueToNoAdjudication() {
+            amendHearingResult(valid, validRequest, 99L, -1L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void amendHearingResultReturns404DueToNoHearing() {
+            amendHearingResult(valid, validRequest, -9L, 2L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void amendHearingResultReturns404DueToNoAdjudicatorOnFile() {
+            amendHearingResult(valid, invalidRequest, -9L, -4L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void amendHearingResultReturns400DueToHearingNotBeingAssociatedWithAdjudication() {
+            amendHearingResult(valid, validRequest, -5L, -4L)
+                .expectStatus().isBadRequest();
+        }
+
+        @Test
+        public void amendHearingResultReturns404DueToNoHearingResultPresent() {
+            amendHearingResult(valid, validRequest, -9L, -4L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void amendHearingResultReturnsSuccess() {
+            amendHearingResult(valid, validRequest, -3001L, -3001L)
+                .expectStatus().isOk()
+                .expectBody()
+                .jsonPath("$.findingCode").isEqualTo(FindingCode.NOT_PROCEED.name())
+                .jsonPath("$.pleaFindingCode").isEqualTo(PleaFindingCode.GUILTY.name());
+        }
```

Comment:
same, also do a GET to prove the data was actually written to database

Reply:
We don't have getHearingResult method, so cannot use it. Saving to DB operation is checked by service test using mock (so covered by test). Should I insert direct DB read here?